newIDE/app/src/Debugger/Inspectors/VariablesContainerInspector.js

```diff
@@ -37,22 +44,33 @@ const transform = variablesContainer => {
   return mapValues(variablesContainer._variables.items, transformVariable);
 };
 
-const handleEdit = (edit, { onCall, onEdit }: Props) => {
+const constructPath = (editPath: string[]) => {
+  const path = [`get(${editPath.shift()})`];
+
+  let skip = false;
+  for (const variableName of editPath) {
+    // Skip every second key as it is the "value"
+    // key which is displayed only for better user
+    // experience but doesn't really exists
+    skip = !skip;
+    if (skip) continue;
+
+    path.push(`getChild(${variableName})`);
+  }
+
+  return path;
+};
+
+const handleEdit = (edit, { onCall, onEdit, variablesContainer }: Props) => {
   // Reconstruct the path to the variable to edit
-  const path = ['_variables', 'items'];
-  edit.namespace.forEach(variableName => {
-    path.push(variableName);
-    path.push('_children');
-  });
-  path.push(edit.name);
+  const path = constructPath(edit.namespace);
 
-  // Guess the type of the new value (number or string)
-  if (parseFloat(edit.new_value).toString() === edit.new_value) {
-    path.push('setNumber');
-    onCall(path, [parseFloat(edit.new_value)]);
-  } else {
-    path.push('setString');
-    onCall(path, ['' + edit.new_value]);
+  if (edit.name === 'type') {
+    path.push('castTo');
+    onCall(path, [edit.new_value]);
```

Comment:
Might be a good idea to make a check on the new_value. If it's not a proper type, we should I think `return false`?